### ISSUE TITLE
Adding release git tags

### ISSUE THREAD

**Issue by r-chris** [thedirtyfew/dash-leaflet#28]
Hi - thanks for this amazing project! It's great to see and seems to work very well already.
I believe this is the accommodating PyPI package: https://pypi.org/project/dash-leaflet/

I noticed you are using versioning on PyPI, but not in your GitHub repo. For reproducibility - if you don't mind - would it be possible to add tags to the Git repo - so that we know which version of the code is released on PyPI?

I'm happy to contribute some GitHub Actions PR if that's something you would be interested in?

**Comment by emilhe** [comment]
Thanks! Yes, that is the pypi package. During the last releases, i was actually starting to think along those lines, i.e. creating tags on git. Since the javascript source is not distributed as part of the python package, it is pretty tedious to tract changes with my current setup. Could you elaborate on what you mean by "contribute some GitHub Actions"? I haven't used GitHub Actions before, but from some quick googling it seems to have to do with automation.

My current build work-flow is simply,

* Bump the version number manually in `package.json`, e.g. to 0.0.18
* Build with npm: `npm run build:all`
* Build pypi package: `python setup.py sdist`
* Push to pypi: `python3 -m twine upload dist/dash_leaflet-0.0.18.tar.gz`

I would be happy for any suggestions for improvements, either in as a PR or as a comment.

**Comment by r-chris** [comment]
Yes, GitHub Actions allow you to automate things - for example the release / deployment of your code - there are plenty of alternatives as well (e.g. CircleCI / TravisCI / etc). Instead of doing those steps "by hand", you could automate them.

For example on most of our projects, we tag the git repository when we want to do a release. We then push that tag to GitHub when launches a GitHub Action to build the code and deploy it. It's also a great way to add some basic testing.

In terms of your build workflow it could look something like this:

- Bump the version number manually in `package.json`, e.g. to 0.0.18
- Commit that change with `git commit -a -m "Updating package.json to v0.0.18"`
- Tag the release with `git tag -a v0.0.18` and then `git push --tags`
- _Let the GitHub Action do the build and release_

**Comment by r-chris** [comment]
I started with some example workflows here: https://github.com/r-chris/dash-leaflet/tree/master/.github/workflows

I seem to be missing some setup steps, though. Could you possibly point out some additional npm initialization that needs to happen when compiling from scratch?

On `npm install` I currently get:

```
  File ".../dash-leaflet/dash_leaflet/__init__.py", line 10, in <module>
    from ._imports_ import *
ModuleNotFoundError: No module named 'dash_leaflet._imports_'
```

**Comment by emilhe** [comment]
Thanks! I have been working with CI before, but not GitHub Action. I'll take a deeper dive one of the coming days.

About the error, as far as i remember, it can just be ignored. After running `npm install` (and getting the error), you should be able to run `npm run build:all`.

**Comment by r-chris** [comment]
ok, great - I think you are right even though that's not ideal for a CI workflow, because it'll just stop on that error - unless we force ignore that ... I'll give that a quick shot

**Comment by r-chris** [comment]
I changed the PR from Draft to "ready" - feel free to play around with it!

**Comment by emilhe** [comment]
I have now merged the pull request. I had to make one addition to make it work,

* Add "pip install -r requirements" to the "python-publish.yml". Otherwise, the dash build process fails.

I had some issues triggering the publish job when pushing tags via cmd, but creating releases (and tags at the same time) using the GitHub browser GUI, everything seems to work. Thanks! Hence, i guess my new workflow will be,

* Bump the version number manually in package.json, e.g. to 0.0.19
* Commit that change with git commit -a -m "Updating package.json to v0.0.19"
* Create a release (and a tag along with it) in the GitHub browser GUI. 
_Let the GitHub Action do the build and release_

Let me know if you have other suggestions to improve the workflow :)
